Katello/katello
# Fixes #32460,#33375 speed up yum filter copy and stop debs from failing filtered CV publishes

This PR speeds up content  view version publishing by only using the multi-copy tasks if yum depsolving is enabled.  If it isn't, we can safely use the "old" method of copying content.  This allows for better concurrency, and yum repositories that are not affected by filters will simply have their repository versions copied.

I also moved to using the RPM modify API instead of the copy API because there is a bug where the copy API does not allow the Pulp workers to work concurrently.  This should also help filtered CV publishing to be a bit faster without depsolving.
Note: I need to test this to check that concurrency is going on.

This PR also fixes an issue where deb repositories in filtered CVVs caused failures due to no `copy_content_for_source` support.

TODO:

- [x] Check that two content view publishes can run at the same time using the modify API.

To test:

Bug [32460](https://projects.theforeman.org/issues/32460)

1) Check that the "Multi" copy tasks aren't used for content view publishes with dependency solving disabled.
2) Check that the content view filtering is still accurate.
3) Check that incremental update is still accurate.  Incremental update still relies on the Multi copy tasks.
4) Check that repositories without filters in content view versions are only having repository versions copied over, not actual content.
  -> You can tell by checking that `Actions::Pulp3::Repository::CopyContent` was not run for that repository during the publish.

Bug [33375](https://projects.theforeman.org/issues/33375)

1) Add a deb repo to a content view that has any filter assigned to it. Ensure the content view publish does not fail.
